### PR TITLE
Minor: Don’t send the DeleteTopicsRequest for the invalid topic names

### PR DESCRIPTION
**Pull request by chia7712**
The invalid topic name is already handled locally so it is unnecessary to send the DeleteTopicsRequest. This PR adds a count to MockClient for testing.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by hachikuji** [comment]
cc @cmccabe

**Comment by cmccabe** [comment]
LGTM.  Thanks, @chia7712

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java [hunk 1]
```diff
@@ -225,27 +225,34 @@ public void testInvalidTopicNames() throws Exception {
             env.kafkaClient().setNode(env.cluster().controller());
 
             List<String> sillyTopicNames = Arrays.asList(new String[] {"", null});
+            int originRequestCount = env.kafkaClient().requestCount();
             Map<String, KafkaFuture<Void>> deleteFutures =
                 env.adminClient().deleteTopics(sillyTopicNames).values();
             for (String sillyTopicName : sillyTopicNames) {
                 assertFutureError(deleteFutures.get(sillyTopicName), InvalidTopicException.class);
             }
+            assertEquals(originRequestCount, env.kafkaClient().requestCount());
```
Comment (hachikuji) [inline clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java]: Maybe we could make this assertion precise? I think we expect the request count to be 0.

File: clients/src/test/java/org/apache/kafka/clients/MockClient.java [hunk 2]
```diff
@@ -461,6 +462,7 @@ public ClientRequest newClientRequest(String nodeId, AbstractRequest.Builder<?>
     @Override
     public ClientRequest newClientRequest(String nodeId, AbstractRequest.Builder<?> requestBuilder, long createdTimeMs,
                                           boolean expectResponse, RequestCompletionHandler callback) {
+        requestCount.incrementAndGet();
```
Comment (hachikuji) [inline clients/src/test/java/org/apache/kafka/clients/MockClient.java]: When do we decrement this?

**Review by hachikuji — COMMENTED** [review]
Thanks for the PR. Left a couple comments.

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java (hunk 1 above)
Comment (chia7712) [reply clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java]: Ya, the expected value of count is zero currently. However, we can't assume the initial count won't be changed in the future. For example, we plan to share the AdminClientUnitTestEnv (or MockClient) between all UTs to speedup the tests.

File: clients/src/test/java/org/apache/kafka/clients/MockClient.java (hunk 2 above)
Comment (chia7712) [reply clients/src/test/java/org/apache/kafka/clients/MockClient.java]: Does it need a way to decrement the count of a class used for testing?

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java (hunk 1 above)
Comment (hachikuji) [reply clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java]: Not sure I follow. What would be the benefit of sharing a MockClient?

File: clients/src/test/java/org/apache/kafka/clients/MockClient.java (hunk 2 above)
Comment (hachikuji) [reply clients/src/test/java/org/apache/kafka/clients/MockClient.java]: I may have misunderstood the intent. I thought it was meant to count in-flight requests, but seems it is a total count instead?

File: clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java (hunk 1 above)
Comment (chia7712) [reply clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java]: bq. What would be the benefit of sharing a MockClient?
Just saying (smile) 

I'm fine to change the assertion.

File: clients/src/test/java/org/apache/kafka/clients/MockClient.java (hunk 2 above)
Comment (chia7712) [reply clients/src/test/java/org/apache/kafka/clients/MockClient.java]: bq. seems it is a total count instead?
yep. used only for testing so it should be fine without decrement call

Comment (cmccabe) [reply clients/src/test/java/org/apache/kafka/clients/MockClient.java]: I think a total count is fine.  Maybe name it `totalRequestCount` to make it obvious that that is what it is.

As @hachikuji commented, there is really no need to share a `MockAdminClient` across multiple tests.  The mock client is extremely lightweight-- it doesn't manage any threads or sockets.

Comment (hachikuji) [reply clients/src/test/java/org/apache/kafka/clients/MockClient.java]: Yes, `totalRequestCount` sounds clearer. We might also consider setting it back to 0 in `MockClient.reset`.

Comment (chia7712) [reply clients/src/test/java/org/apache/kafka/clients/MockClient.java]: Thanks for all reviews. Will address the following comments
1) requestCount -> totalRequestCount
2) set totalRequestCount to zero in MockClient.reset

**Review by hachikuji — APPROVED** [review]
LGTM. Thanks for the patch!

**Pull request merged**